Emby.Server.Implementations/Plugins/PluginManager.cs

```diff
@@ -427,9 +432,72 @@ public async Task<bool> GenerateManifest(PackageInfo packageInfo, Version versio
                 ImagePath = imagePath
             };
 
+            var metafile = Path.Combine(Path.Combine(path, MetafileName));
+            if (File.Exists(metafile))
+            {
+                var data = File.ReadAllBytes(metafile);
+                var localManifest = JsonSerializer.Deserialize<PluginManifest>(data, _jsonOptions) ?? new PluginManifest();
+
+                // Plugin installation is the typical cause for populating a manifest. Activate.
+                localManifest.Status = status == PluginStatus.Disabled ? PluginStatus.Disabled : PluginStatus.Active;
+
+                if (!Equals(localManifest.Id, manifest.Id))
+                {
+                    _logger.LogError("The manifest ID {LocalUUID} did not match the package info ID {PackageUUID}.", localManifest.Id, manifest.Id);
+                    localManifest.Status = PluginStatus.Malfunctioned;
+                }
+
+                if (localManifest.Version != manifest.Version)
+                {
+                    _logger.LogWarning("The version of the local manifest was {LocalVersion}, but {PackageVersion} was expected. The value will be replaced.", localManifest.Version, manifest.Version);
+
+                    // Correct the local version.
+                    localManifest.Version = manifest.Version;
+                }
+
+                // Reconcile missing data against repository manifest.
+                ReconcileManifest(localManifest, manifest);
+
+                manifest = localManifest;
+            }
+            else
+            {
+                _logger.LogInformation("No local manifest exists for plugin {Plugin}. Populating from repository manifest.", manifest.Name);
+            }
+
             return SaveManifest(manifest, path);
         }
 
+        /// <summary>
+        /// Resolve the target plugin manifest against the source. Values are mapped onto the
+        /// target only if they are default values or empty strings. ID and status fields are ignored.
+        /// </summary>
+        /// <param name="baseManifest">The base <see cref="PluginManifest"/> to be reconciled.</param>
+        /// <param name="projector">The <see cref="PluginManifest"/> to reconcile against.</param>
+        private void ReconcileManifest(PluginManifest baseManifest, PluginManifest projector)
+        {
+            var ignoredFields = new string[]
+            {
+                nameof(baseManifest.Id),
+                nameof(baseManifest.Status)
+            };
+
+            foreach (var property in baseManifest.GetType().GetProperties())
```

Comment:
I know this is more "future-proof" but I am not a fan of iterating through all the properties. This should be specific in the properties it maps

Reply:
Perhaps I could just move the reflection aspect into the tests so that new unmapped properties are detected early.  I'll take a look.